chat-android/src/main/java/com/ably/chat/RoomOptions.kt

```diff
@@ -1,113 +1,187 @@
 package com.ably.chat
 
+import com.ably.chat.annotations.ChatDsl
 import io.ably.lib.types.ChannelMode
 import io.ably.lib.types.ChannelOptions
 
 /**
  * Represents the options for a given chat room.
  */
-public data class RoomOptions(
+public interface RoomOptions {
     /**
      * The presence options for the room. To enable presence in the room, set this property. You may
-     * use [RoomOptionsDefaults.presence] to enable presence with default options.
+     * use `rooms.get("ROOM_NAME") { presence() }` to enable presence with default options.
      * @defaultValue undefined
      */
-    val presence: PresenceOptions? = null,
+    public val presence: PresenceOptions?
 
     /**
      * The typing options for the room. To enable typing in the room, set this property. You may use
-     * [RoomOptionsDefaults.typing] to enable typing with default options.
+     * `rooms.get("ROOM_NAME") { typing() }`to enable typing with default options.
      */
-    val typing: TypingOptions? = null,
+    public val typing: TypingOptions?
 
     /**
      * The reactions options for the room. To enable reactions in the room, set this property. You may use
-     * [RoomOptionsDefaults.reactions] to enable reactions with default options.
+     * `rooms.get("ROOM_NAME") { reactions() }` to enable reactions with default options.
      */
-    val reactions: RoomReactionsOptions? = null,
+    public val reactions: RoomReactionsOptions?
 
     /**
      * The occupancy options for the room. To enable occupancy in the room, set this property. You may use
-     * [RoomOptionsDefaults.occupancy] to enable occupancy with default options.
+     * `rooms.get("ROOM_NAME") { occupancy() }` to enable occupancy with default options.
      */
-    val occupancy: OccupancyOptions? = null,
-) {
+    public val occupancy: OccupancyOptions?
+
     public companion object {
         /**
          * Supports all room options with default values
          */
-        public val default: RoomOptions = RoomOptions(
-            typing = TypingOptions(),
-            presence = PresenceOptions(),
-            reactions = RoomReactionsOptions(),
-            occupancy = OccupancyOptions(),
-        )
+        public val AllFeaturesEnabled: RoomOptions = buildRoomOptions {
```

Comment:
Seems this will change as a part of single channel migration.
All feature will be enabled by default, so we don't need to provide explicit option for the same.
This is because, combining multiple features into single channel allows to have all features enabled without extra overhead.